### ISSUE TITLE
Feedback: 'Persistence'

### ISSUE THREAD

**Issue by stevebroshar** [Particular/docs.particular.net#4070]
Feedback for 'Persistence' https://docs.particular.net/persistence/

Location in GitHub: https://github.com/Particular/docs.particular.net/blob/master/persistence/index.md

I googled "nservicebus what is persistence used for" which brought me to this page. But, this page doesn't really answer that question. Clearly, nsb uses a database, It is used in various nsb features. It supports different DB servers. But what does nsb store in there? How is the information used? When does nsb write to the DB? When does it read from the DB? I want to know these things since they affects my system design. NSB is too black-box in this (and other) respects. 

In particular (no pun in tended), I have endpoint setup issues that I think might be related to what's stored in the DB. But, it's guesswork since I don't know how nsb uses the DB.

In general nsb is great when it's working, but when it's not, it's a big black box ... huge frustration since I don't know how to fix, how to debug, what to try. FWIW, we are thinking of scrapping nsb due to it's black-boxiness and flakiness.

**Comment by mikeminutillo** [comment]
@stevebroshar the page has a list of features that rely on the persistence feature. Would adding a brief explanation of what those features use the persistence for help or do you need something more in depth?

**Comment by danielmarbach** [comment]
We have outlined the features that rely on the persistence and in general in most transports we have introduce native ways to for example timeouts as well as subscriptions (some work is still ongoing there). Recently we also [created guidance in how to select a persistence](https://docs.particular.net/persistence/selecting) that should give a good understanding of NSB uses the storage. Closing this for now. Should you have more feedback on how we can improve things don't hesitate to bring it to our attention and we will see what we can do.
